cli/cli.go

```diff
@@ -37,14 +38,49 @@ func printVersion(c *cli.Context) {
 }
 
 func action(c *cli.Context) {
-	if os.Getenv(plugins.PluginInputPluginModeKey) == string(plugins.TriggerMode) {
-		sendAnalytics()
+	if os.Getenv(plugins.PluginInputPluginModeKey) != string(plugins.TriggerMode) {
+		log.Errorf("Required envs not set: only Bitrise CLI is intended to send build run analytics")
+
+		if err := cli.ShowAppHelp(c); err != nil {
+			failf("Failed to show help, error: %s", err)
+		}
+		return
+	}
+
+	if enabled, err := isAnalyticsEnabled(); err != nil {
+		failf("Failed to check if analytics enabled: %s", err.Error())
+	} else if !enabled {
+		log.Debugf("Build run analytics disabled, terminating...")
+		return
+	}
+
+	if warn, err := checkFormatVersion(os.Getenv(plugins.PluginInputFormatVersionKey), models.Version); err != nil {
+		failf(err.Error())
+	} else if len(warn) > 0 {
+		log.Warnf(warn)
+	}
+
+	var t SourceType
+	if available, err := hasContent(os.Stdin); err != nil {
+		failf("Failed to check if analytics enabled: %s", err.Error())
+	} else if available {
+		log.Debugf("stdin payload provided")
+		t = StdinSource
```

Comment:
NOTE: I suggest to set StdIn source struct that implements PayloadSource here

Reply:
why?